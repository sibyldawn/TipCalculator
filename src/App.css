.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 80px;
}

.App-header {
  background: rgb(249,244,244);
background: linear-gradient(221deg, rgba(249,244,244,1) 0%, rgba(86,233,114,1) 35%, rgba(0,212,255,1) 100%);
  height: 150px;
  width: 100%;
  padding: 20px;
  color: white;
  position: fixed;
  top:0;
}

.App-title {
  font-size: 1.5em;
}

.App-intro {
  font-size: large;
}

@keyframes App-logo-spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.wrapper{
  display: flex;
  justify-content: space-around;
}


.tip-box{
  flex:1;
  background:rgba(206, 248, 247  
  ,0.5);
  height: 70vh;
  min-width: 300px;
  position: absolute;
  top: 200px;
  margin-left:auto;
  font-size: 25px;
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
  color: slategrey;
}

button{
margin-top: 10px;
 width: 300px;
 height: 50px;
 background: #36F71C;
 font-size: 25px;
 font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
 color: white;
 position: relative;
 bottom: 0;
}

input{
  font-size: 36px;
  text-align: center;
  width: 200px;
  border: none;
  border-bottom: solid 1px gray;
  background: transparent;
}

@media(min-width: 600px){
  .tip-box{
    width: 500px;
  }
  button{
    width: 500px;
  }
}