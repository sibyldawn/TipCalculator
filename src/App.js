import React, { Component } from 'react';
import './App.css';

class App extends Component {
  constructor(){
    super()

    this.state = {
      bill: 0,
      tax: 0,
      tipPercent: 0,
      total: 0,
    }
  }

  handleBillChange(bill){
    this.setState({
         bill
    },this.calculateTax())
  }
  
  calculateTax(){
    const { bill } = this.state;

    let tax = bill * 0.86
    
    return(
      this.setState({
           tax: tax.toFixed(2)
      })
    )
  }


  handleTipChange(tipPercent){
    this.setState({
         tipPercent
    },this.calculateTotal)
  }
  

  calculateTotal(){
    const {bill,tax,tipPercent} = this.state;
    
    let subtotal = +bill + +tax;
    console.log("subtotal", subtotal);
    let tipAmount = (subtotal*(tipPercent/100)).toFixed(2)
    console.log("tip", tipAmount);
    let total = +subtotal + +tipAmount
    console.log(total)
    return(
      this.setState({
        total: total
      })
    )
  }



  render() {
   const { tax, total} = this.state
    return (
      <div className="App">
        <header className="App-header">
          <img src="https://www.socialenterprisemark.org.uk/wp-content/uploads/2016/07/Dollar-icon.png" className="App-logo" alt="logo" />
          <h1 className="App-title">Tip Calculator</h1>
        </header>
        <div className="wrapper">
        <div className="tip-box">
          <p>Bill:</p><span>$</span><input type="number" onChange={e => this.handleBillChange(e.target.value)}/> 
          <p>Tax:</p><h3>${tax}</h3>
          <p>Tip Percentage:</p><input type="number" onChange={e => this.handleTipChange(e.target.value)}/><span>%</span>
          <p>Total Bill:</p><h3>${total}</h3>
         <div> <button onClick={()=> alert(`Received Payment: $${total}`)}>Pay Now</button> </div>
        </div>
        </div>  
      </div>
    );
  }
}

export default App;
