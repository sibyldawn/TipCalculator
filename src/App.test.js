import React from 'react';
import App from './App';
import Enzyme,{ shallow } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({adapter : new EnzymeAdapter() });

it('renders without issues', () => {
  const wrapper = shallow(<App/>);
  const appComponent = wrapper.find("[data-test='component-app']")
  expect(appComponent.length).toBe(1)
})

it('matches the snapshot',() => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot();
})
