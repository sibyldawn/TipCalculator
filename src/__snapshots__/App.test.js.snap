// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<div
  className="App"
  data-test="component-app"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="https://www.socialenterprisemark.org.uk/wp-content/uploads/2016/07/Dollar-icon.png"
    />
    <h1
      className="App-title"
    >
      Tip Calculator
    </h1>
  </header>
  <div
    className="wrapper"
  >
    <div
      className="tip-box"
    >
      <p>
        Bill:
      </p>
      <span>
        $
      </span>
      <input
        onChange={[Function]}
        type="number"
      />
      <p>
        Tax:
      </p>
      <h3>
        $
        0
      </h3>
      <p>
        Tip Percentage:
      </p>
      <input
        onChange={[Function]}
        type="number"
      />
      <span>
        %
      </span>
      <p>
        Total Bill:
      </p>
      <h3>
        $
        0
      </h3>
      <div>
         
        <button
          onClick={[Function]}
        >
          Pay Now
        </button>
         
      </div>
    </div>
  </div>
</div>
`;
